token-metadata/program/src/utils.rs

```diff
@@ -867,20 +891,45 @@ pub fn process_create_metadata_accounts_logic(
     )?;
 
     let mut metadata = Metadata::from_account_info(metadata_account_info)?;
+    let compatible_data = data.to_v1();
     assert_data_valid(
-        &data,
+        &compatible_data,
         &update_authority_key,
         &metadata,
         allow_direct_creator_writes,
         update_authority_info.is_signer,
         false,
     )?;
 
+    let mint_decimals = get_mint_decimals(mint_info)?;
+
     metadata.mint = *mint_info.key;
     metadata.key = Key::MetadataV1;
-    metadata.data = data;
+    metadata.data = Data {
+        name: data.name,
+        symbol: data.symbol,
+        uri: data.uri,
+        seller_fee_basis_points: data.seller_fee_basis_points,
+        creators: data.creators,
+    };
     metadata.is_mutable = is_mutable;
     metadata.update_authority = update_authority_key;
+    assert_valid_use(&data.uses, &None)?;
+    metadata.uses = data.uses;
+    assert_collection_update_is_valid(&None, &data.collection)?;
+    metadata.collection = data.collection;
+    if add_token_standard {
+        let token_standard = if is_edition {
+            TokenStandard::Edition
+        } else if mint_decimals == 0 {
+            TokenStandard::SemiFungible
+        } else {
+            TokenStandard::Fungible
+        };
+        metadata.token_standard = Some(token_standard);
+    } else {
+        metadata.token_standard = None;
```

Comment:
Under no circumstance should this be none going forward. You can always figure out what it is given mint info.